cargo/private/cargo_utils.bzl

```diff
@@ -151,18 +151,41 @@ def _resolve_repository_template(
 
     return template
 
-def get_rust_tools(cargo_template, rustc_template, host_triple, version):
+def cargo_home_path(repository_ctx):
+    """Define a path within the repository to use in place of `CARGO_HOME`
+
+    Args:
+        repository_ctx (repository_ctx): The rules context object
+
+    Returns:
+        path: The path to a directory to use as `CARGO_HOME`
+    """
+    return repository_ctx.path(".cargo_home")
+
+def get_rust_tools(cargo_template, rustc_template, host_triple, version, repository_ctx):
     """Retrieve `cargo` and `rustc` labels based on the host triple.
 
     Args:
         cargo_template (str): A template used to identify the label of the host `cargo` binary.
         rustc_template (str): A template used to identify the label of the host `rustc` binary.
         host_triple (struct): The host's triple. See `@rules_rust//rust/platform:triple.bzl`.
         version (str): The version of Cargo+Rustc to use.
+        repository_ctx (repository_ctx): The rule's context object.
```

Comment:
I combine the `get_rust_tools` functions multiple files into this one. One of them required `repository_ctx`. Given that multiple attrs from it are used, I'd prefer to leave that refactor to a future commit since I think it will require non-trivial plumbing. Would that be okay?